### DISCUSSION TITLE
Soraban random library

### DISCUSSION THREAD

**Discussion by vinamogit** [stellar/sorobanathon#112] [General]
Here is an attempt to provide some level of randomness into a contract.

It is based on the Rust rand common trait and implements [SmallRng](https://docs.rs/rand/0.5.2/rand/rngs/struct.SmallRng.html) for which the seed is derived from contract and ledger context. The underlying implementation is Xoshiro128PlusPlus.

[soroban-rand](https://github.com/vinamogit/soroban-rand)

**Comment by hanseartic** [comment]
🙏🏼 Thank you for this! It even [empowered](https://github.com/hanseartic/sorobanathon/blob/main/trait_contract/Cargo.toml#L18) me to submit a contract on my own – sorobanathon'ception 🕶️

**Reply by vinamogit** [reply]
Thanks! and thank you for the PR, too.

**Comment by achallagundla** [comment]
Heya,
Congratulations! Your contribution to Sorobanathon: First Light has been accepted! 🎉
Your submission may qualify for a reward of 1400 XLM! One last step: to receive that reward, please fill out this [form](https://stellarform.typeform.com/Sorobanathon) completely and accurately.

Please let us know if you have any questions. (You can also message me on discord @ anuxhya)

All the best,
Anuhya

PS: No need to submit the form if you already have, just update me on discord if your wallet address has changed.
